[Modal] 로그인/회원가입 및 상세페이지에 필요한 모달 컴포넌트 완성

hanana1253 opened the pull request:
## Related Issue
#14 

## Done
- 모달(Dialog) 컴포넌트를 만들었습니다.
- 키보드 트래핑 처리
- 스크롤 처리

## 비고

hanana1253 committed:
chore: Add tabbable util module

야무강사님의 tabbable util을 가져왔습니다.

TODO: 키보드 트래핑 구현

hanana1253 committed:
feat: Make modal dialog component

I made modal component.

TODO: Try nested modal dialog...

hanana1253 committed:
feat: Add scroll feature and responsive background image to modal dialog

I made scroll on background to be fixed when modal dialog is open.
Background image is included in modal dialog.

hanana1253 committed:
feat: Add dialog component and its styling

- index.html에 #dialog div요소 추가
- Dialog 컴포넌트 추가
- Dialog example 추가
- Dialog styling scss 파일 추가
- asset default image 추가

Merge branch 'feature/dialog' into develop

hanana1253 committed:
Merge branch 'develop' of https://github.com/TeamCooks/hanspoon into develop

hanana1253 committed:
fix: Add role attribute to img tag in Dialog.Image(BackgroundImage)

I added role attribute to img tag for accessibility and semantic use.

hanana1253 committed:
chore: Move SignIn and SignUp components from pages directory to components directory

They were in wrong path...

TODO: put SignIn and SignUp inside modal dialog

hanana1253 committed:
chore: Add new components' paths to components/index.js

hanana1253 committed:
fix: Fix SignIn/SignUp paths in App.js

I moved files and forgot to fix paths which import the files.
Sorry.

hanana1253 committed:
chore: Make 404 page.

TODO: Fix main component padding and add root variable of header & footer height

hanana1253 commented:
feature finish한 후 제 develop에서 push 했더니 모달창과 상관없는 커밋도 여기 올라가버렸습니당. 
#29 와 관련된 page not found 페이지 커밋입니다.
나중에 PR을 별도로 만드는 법이 있는지 울군샘에게 물어볼게요.

hanana1253 committed:
feat: Make Search Component

- 10개의 검색결과를 가져옵니다.

TODO: pagination and loading spinner

Fine-Tuner commented:
> feature finish한 후 제 develop에서 push 했더니 모달창과 상관없는 커밋도 여기 올라가버렸습니당. #29 와 관련된 page not found 페이지 커밋입니다. 나중에 PR을 별도로 만드는 법이 있는지 울군샘에게 물어볼게요.

넵! 알게되시면 노션에 적어주시거나 공유부탁드립니다. 고생하셨습니다.

hanana1253 committed:
feat: Add search feature and pagination with caching in state.

- search한 결과를 10개씩 페이지에 렌더링합니다.
- 이미 방문한 페이지 정보는 caching 처리하는 state

TODO:
- custom 훅으로 fetching 빼기
- 로딩 컴포넌트가 안 뜨고 있는데 처리해주기

hanana1253 committed:
feat: Put Card component inside Search Page

- Card component 합체!

TODO:
- Loading component 안 나오는 문제 해결하기
- Pagination 예외처리 (currentPage가 1 미만일 때 등)
- 커스텀 훅으로 빼기

hanana1253 committed:
fix: Add routing path in App.js and fix typo in Search component

Added Routing path for Search page and fixed typo

skojphy commented on src/App.js:
```diff
@@ -25,6 +26,7 @@ function App() {
         { path: 'detail', element: <Detail /> },
         { path: 'signin', element: <SignIn /> },
         { path: 'signup', element: <SignUp /> },
+        { path: 'search/:keyword', element: <Search /> },
```
keyword가 공백을 포함할 경우에 경로를 _ 등으로 구분한 다음
API를 호출할 때 다시 공백으로 바꿔주는 것이 필요하지 않을지 확인해야 할 것 같습니다.

hhhyyo reviewed (approved):
확인했습니다

Fine-Tuner reviewed (approved):
좋습니다!

skojphy reviewed (commented):
@hhhyyo @hanana1253

hanana1253 reviewed (commented):
Thank you
